### ISSUE TITLE
vmagent: shard targets by the body size in cluster mode

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#3209]
**Is your feature request related to a problem? Please describe.**
Some targets may expose thousands and millions of time series. This makes such a target very heavy and may exhaust resources of the vmagent which is scraping it.

**Describe the solution you'd like**
Would be great if vmagent could shard not only targets (vmagent1 scrapes 1-50 targets, vmagent2 scrapes 51-100 targets) in cluster mode. But also to shard the heavy targets as well (vmagent1 scrapes 50% of heavy-target1 body, and vmagent2 scrapes rest 50% of heavy-target1 body). This would split the "work" more evenly between vmagents in the cluster mode and result into more balanced resource consumption.

**Describe alternatives you've considered**
Modifying targets to not expose a too big number of metrics.

**Comment by leo-minorui** [comment]
any updates？

**Comment by leo-minorui** [comment]
Is it the same as the prometheus sharding mechanism on current version?

**Comment by hagen1778** [comment]
> Is it the same as the prometheus sharding mechanism on current version?

No. Prometheus-like sharding is [already supported](https://docs.victoriametrics.com/vmagent.html#scraping-big-number-of-targets). Sharding mentioned in this ticket is meant to shard not only within list of address, but also within one address.

**Comment by leo-minorui** [comment]
> > Is it the same as the prometheus sharding mechanism on current version?
> 
> No. Prometheus-like sharding is [already supported](https://docs.victoriametrics.com/vmagent.html#scraping-big-number-of-targets). Sharding mentioned in this ticket is meant to shard not only within list of address, but also within one address.

Great！！！ Looking forward more news
